raddar/api/api_v1/endpoints/webhooks.py

```diff
@@ -0,0 +1,43 @@
+from typing import Any, List
+
+from fastapi import APIRouter, Body, Depends, Header, Request
+from sqlalchemy.orm import Session
+
+from raddar.core import contexts, dependencies
+from raddar.crud import crud
+from raddar.schemas import schemas
+from raddar.core.settings import settings
+from raddar.api import deps
+from raddar.lib.managers.detect_secrets_manager import get_project_secrets
+from raddar.lib.managers.repository_manager import get_branch_name
+
+
+router = APIRouter()
+
+
+@router.post("/github", dependencies=[Depends(dependencies.valid_github_webhook)])
+def scan_github_project(payload: dict, db: Session = Depends(deps.get_db)):
+    with contexts.clone_repo(
+        project_dir=settings.PROJECT_RESULTS_DIRNAME,
+        project_name=payload["repository"]["full_name"],
+        ref_name=payload["ref"],
+    ) as (repo, temp_dir):
+        analyze_returned = crud.create_analyze(
+            db=db,
+            project=schemas.ProjectBase(name=payload["repository"]["full_name"]),
+            analyze=schemas.AnalyzeBase(branch_name=get_branch_name(payload["ref"])),
+            ref_name=repo.commit("HEAD").hexsha,
+            origin="github-webhook"
```

Comment:
Maybe this `origin` could be documented in an Enum, that way it's typed and you don't rely on passing strings around (or you can keep it a string and document which values are accepted, see a later comment for this).

Reply:
Literal is used in function definition